### PR TITLE
Prepend custom iptables chains instead of appending

### PR DESCRIPTION
**Pull request by tgraf**
Fixes: #5766

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/5767)

### CODE REVIEW

**Comment by tgraf** [comment]
test-me-please

**Review by ianvernon — APPROVED** [review]
Approving for docs.

**Comment by ianvernon** [comment]
Can you provide an example of testing done using this option? I.e., when deployed with this option, does it behave nicely with K8s / kube-proxy?

**Comment by tgraf** [comment]
> Can you provide an example of testing done using this option? I.e., when deployed with this option, does it behave nicely with K8s / kube-proxy?

I tested this manually by switch back and forth using the option and not using the option. There was no change in the iptables rules themselves. This is done on user request so we will also get direct testing from Svintoo on slack.

**Review by raybejjani — APPROVED** [review]
A basic sanity check runtime test that checks that anything works might be worthwhile here. I'm not sure what the impact here is, since I don't fully know the normal IP chains. Would it be worth adding a doc section explaining this option?
I vaguely remember that nodeport relies on iptable rules from kubernetes and I assumed those happened before our rules. Will this break nodeport?

**Comment by tgraf** [comment]
> I vaguely remember that nodeport relies on iptable rules from kubernetes and I assumed those happened before our rules. Will this break nodeport?

I've done the testing manually and there was no change in the effective rules.

Thinking more about this. I've decided to just change the default instead so the next CI run will cover testing with the new behavior.

**Comment by tgraf** [comment]
test-me-please

green build with rebased PR and option turned off

**Comment by tgraf** [comment]
@raybejjani @ianvernon I've reverted your reviews as I've changed the default value of the option. Please see the commit description of the 2nd commit for the justification.

**Comment by tgraf** [comment]
test-me-please

EDIT: forgot to update the cmdref with the new default

File: Documentation/cmdref/cilium-agent.md [hunk 1]
```diff
@@ -75,6 +75,7 @@ cilium-agent
       --pprof                                       Enable serving the pprof debugging API
       --prefilter-device string                     Device facing external network for XDP prefiltering (default "undefined")
       --prefilter-mode string                       Prefilter mode { native | generic } (default: native) (default "native")
+      --prepend-iptables-chains                     Prepend custom iptables chains instead of appending
```
Comment (aanm) [inline Documentation/cmdref/cilium-agent.md]: add the default value of this flag in the help message.

**Comment by tgraf** [comment]
test-me-please

**Review by joestringer — APPROVED** [review]

**Review by ianvernon — APPROVED** [review]

**Comment by tgraf** [comment]
test-me-please

**Review by raybejjani — APPROVED** [review]

**Pull request merged**